shared/src/components/Navigation/NavigationMenu/NavigationMenu.tsx

```diff
@@ -0,0 +1,80 @@
+import type { HTMLProps } from "react";
+import React, { useCallback, useState } from "react";
+
+import type { PropsWithSpread } from "@canonical/react-components";
+import classNames from "classnames";
+
+import { useClickOutside } from "../../../utils";
+import NavigationLink from "../NavigationLink";
+import type { GenerateLink, NavMenu } from "../types";
+
+type Props = PropsWithSpread<
+  NavMenu & {
+    generateLink?: GenerateLink;
+  },
+  HTMLProps<HTMLLIElement>
+>;
+
+/**
+ * This component is used internally to display menus inside the Navigation component.
+ */
+const NavigationMenu = ({
+  alignRight,
+  generateLink,
+  items,
+  label,
+  ...props
+}: Props): JSX.Element => {
+  const [isOpen, setIsOpen] = useState(false);
+  const closeMenu = useCallback(() => setIsOpen(false), [setIsOpen]);
+  const [menuRef, menuId] = useClickOutside<HTMLAnchorElement>(closeMenu);
+  return (
+    <li
+      {...props}
+      className={classNames(
+        props.className,
+        "p-navigation__item--dropdown-toggle",
+        {
+          "is-active": isOpen,
+        }
+      )}
+    >
+      {/* TODO: replace anchor with button https://github.com/canonical-web-and-design/maas-ui/issues/3454 */}
```

Comment:
I think this is a different issue, and potentially a Vanilla one instead (i.e. Vanilla should support navigation menus as buttons)

Reply:
Thanks yes, this branch actually fixes that issue! Also, you made me realise that it supports buttons if you give them the class `p-navigation__link` so I've replaced this with a button.